source/common/router/router.cc

```diff
@@ -491,10 +543,12 @@ void Filter::onUpstreamHeaders(Http::HeaderMapPtr&& headers, bool end_stream) {
   if (retry_state_) {
     RetryStatus retry_status = retry_state_->shouldRetry(
         headers.get(), Optional<Http::StreamResetReason>(), [this]() -> void { doRetry(); });
+    const auto upstream_host = upstream_request_->upstream_host_;
```

Comment:
Q: Why capture local var here?

Reply:
`setupRetry` in the next line will clear `upstream_request_`, but we need it in the `if` body. Will add a comment.